src/Core/tests/Unit/ConfigurationTest.php

```diff
@@ -9,30 +9,41 @@
 
 class ConfigurationTest extends TestCase
 {
-    public function testDefaultValues()
+    /**
+     * @dataProvider provideConfiguration
+     */
+    public function testCreate($config, $env, $expected)
     {
-        $config = Configuration::create(['endpoint' => 'foo']);
-        self::assertEquals('foo', $config->get('endpoint'));
-
-        $config = Configuration::create(['endpoint' => '']);
-        self::assertEquals('', $config->get('endpoint'));
-
-        $config = Configuration::create([]);
-        self::assertEquals('https://%service%.%region%.amazonaws.com', $config->get('endpoint'));
-
-        // If a configuration value is set to null, we should use default value.
-        $config = Configuration::create(['endpoint' => null]);
-        self::assertEquals('https://%service%.%region%.amazonaws.com', $config->get('endpoint'));
-
-        $config = Configuration::create([]);
-        self::assertEquals('default', $config->get('profile'));
-
-        putenv('AWS_PROFILE=foo');
-        $config = Configuration::create([]);
-        self::assertEquals('foo', $config->get('profile'));
+        foreach ($env as $key => $value) {
+            putenv("$key=$value");
+        }
+
+        try {
+            $config = Configuration::create($config);
+            foreach ($expected as $key => $value) {
+                self::assertEquals($value, $config->get($key));
+            }
+        } finally {
+            foreach ($env as $key => $value) {
+                putenv($key);
+            }
+        }
+    }
 
-        putenv('AWS_PROFILE');
-        $config = Configuration::create([]);
-        self::assertEquals('default', $config->get('profile'));
+    public function provideConfiguration(): iterable
+    {
+        yield 'simple config' => [['endpoint' => 'foo'], [], ['endpoint' => 'foo']];
+        yield 'empty config' => [['endpoint' => ''], [], ['endpoint' => '']];
+        yield 'default' => [[], [], ['endpoint' => 'https://%service%.%region%.amazonaws.com']];
+        yield 'default when null' => [['endpoint' => null], [], ['endpoint' => 'https://%service%.%region%.amazonaws.com']];
+
+        yield 'default when env missing' => [[], [], ['profile' => 'default']];
+        yield 'fallback env' => [[], ['AWS_PROFILE' => 'foo'], ['profile' => 'foo']];
+        yield 'config priority on env' => [['profile' => 'bar'], ['AWS_PROFILE' => 'foo'], ['profile' => 'bar']];
+
+        yield 'config with env group' => [['accessKeyId' => 'key'], [], ['accessKeyId' => 'key', 'sessionToken' => null]];
+        yield 'config with env group 2' => [['accessKeySecret' => 'secret'], [], ['accessKeySecret' => 'secret', 'accessKeyId' => null, 'sessionToken' => null]];
+        yield 'mix config and env' => [['accessKeyId' => 'key'], ['AWS_SESSION_TOKEN' => 'token'], ['accessKeyId' => 'key', 'sessionToken' => null]];
```

Comment:
This test covers #280:
When setting the configuration `accessKeyId` and exposing an env variable `AWS_SESSION_TOKEN`, the result configuration should have `sessionToken = null`